ocaml/xapi-idl/storage/storage_interface.ml

```diff
@@ -235,6 +235,16 @@ let default_vdi_info =
 type sr_health = Healthy | Recovering | Unreachable | Unavailable
 [@@deriving rpcty]
 
+let sr_health_to_string = function
+  | Healthy ->
+      "healthy"
+  | Recovering ->
+      "recoverying"
+  | Unreachable ->
+      "unreachable"
+  | Unavailable ->
+      "unavailabe"
```

Comment:
There is no need to manually specify the conversion here. Once #5951 is merged, an automatically-generated and up-to-date `sr_health_to_string` (and `of_string`) will be available from Record_util.

Reply:
(If lowercase conversion is preferred, a lowercase `to_string` can be defined in terms of the autogenerated function)

Reply:
Right.... 

Weird that there's two copies of this type - one in the datamodel (producing a list of polymorphic variants), and one here (a strict enum). 

Seems to be introduced in https://github.com/xapi-project/xen-api/commit/040d0ef62bda87a01e50e9293661a16a9f592ebf - is this a common practice to get an Enum type out of a datamodel? Seems error-prone as changes could be introduced into `Sr_stat.health` or here without sync-up

Reply:
probably because the datamodel  type definitions does not support normal variants....

Reply:
Strictly speaking, this is a different IDL/interface from the high-level datamodel. I don't think that any of the interfaces in `ocaml/xapi-idl` depend on the datamodel (from `ocaml/idl`) or other parts of xapi and I think it is best to keep it that way.